### PR TITLE
US-1 Node List

### PR DESCRIPTION
**Pull request by toshydev**

### CODE REVIEW

File: backend/src/test/java/click/snekhome/backend/service/NodeServiceTest.java [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+        List<Node> expected = List.of(node1, node2);
+        //when
+        when(nodeRepo.findAll()).thenReturn(expected);
+        NodeService nodeService = new NodeService(nodeRepo);
+        List<Node> actual = nodeService.list();
+        //then
+        assertEquals(expected, actual);
+        verify(nodeRepo).findAll();
+    }
+}
```
Comment (WE-Kaito) [inline backend/src/test/java/click/snekhome/backend/service/NodeServiceTest.java:30]: 👀
image

File: frontend/src/models.ts [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+    lastUpdated: string,
+    lastAttacked: string,
+    coordinates: Coordinates,
+}
+
+type Coordinates = {
+    latitude: number,
+    longitude: number,
+    timestamp: number
+}
```
Comment (WE-Kaito) [inline frontend/src/models.ts:16]: 😨
image

**Review by WE-Kaito — APPROVED** [review]
Clean Code 🌟
You were probably hired for Phantom Liberty's marketing😏

**Review by StormTrooper-007 — COMMENTED** [review]
Nice one ... in the NodeController you may want you use @RequiredArgsConstructor annotation. Then you would'nt need to instantiate the NodeController.

**Comment by toshydev** [comment]
@StormTrooper-007 Thanks a lot for the review. 
I started to use Lombok less to understand how spring takes care of dependency injection. But you're right. It's a lot less boilerplate code with the `@RequiredArgsConstructor`.

**Pull request merged**